plugins/snowflake/snowsql.go

```diff
@@ -0,0 +1,22 @@
+package snowflake
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/needsauth"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+)
+
+func SnowflakeCLI() schema.Executable {
+	return schema.Executable{
+		Name:      "Snowflake",
+		Runs:      []string{"snowsql"},
+		DocsURL:   sdk.URL("https://docs.snowflake.com/en/user-guide/snowsql.html"),
+		NeedsAuth: needsauth.NotForHelpOrVersion(),
```

Comment:
Good point! I modified the rules so that it opts out of 1Password authentication if any of the following arguments are passed:
```
-a
--accountname
-u
--username
--authenticator
--config
-P
--prompt
```

Descriptions for each of these arguments can be found here: https://docs.snowflake.com/en/user-guide/snowsql-start.html#connection-syntax. Let me know if there seem to be any arguments I missed, or if we shouldn't opt-out of authentication for any of the arguments I included.